core/src/main/java/feign/InvocationHandlerFactory.java

```diff
@@ -33,6 +33,7 @@
   interface MethodHandler {
 
     Object invoke(Object[] argv) throws Throwable;
+    MethodMetadata getMethodMetadata();
```

Comment:
Required to change the return type to the actual type to send on to the decoder. If the return type is `HystrixCommand<User>`, `User` is set as the return type.